e2e/cypress/tests/codebase.spec.ts

```diff
@@ -0,0 +1,121 @@
+    cy.get('[data-cy="codebase title"]').type("Codebase Title");
+    cy.get('[data-cy="codebase description"]').type("Codebase Description");
+    cy.get('[data-cy="codebase replication-text"]').type(
+      "Codebase Replication Text"
+    );
+    cy.get('[data-cy="codebase associated publications"]').type(
+      "Codebase Associated Publications"
+    );
+    cy.get('[data-cy="codebase references"]').type("Codebase References");
+    cy.get('[data-cy="next"]').click();
+    //add images
+    cy.get('[data-cy="add image"]').click(); //add image
+    cy.get('[data-cy="upload-image"]')
+      .first()
+      .selectFile("cypress/fixtures/codebase/codebasetestimage.png", {
+        force: true,
+      });
+    cy.get('button.btn.btn-outline-gray[data-bs-dismiss="modal"]')
+      .get("button")
+      .first()
+      .click({ force: true });
+    cy.get('button.btn.btn-outline-gray[data-bs-dismiss="modal"]')
+      .should("be.visible")
+      .and("not.be.disabled")
+      .first()
+      .click({ force: true });
+    cy.get("body").click(0, 0);
+    cy.get("body").click(0, 0); //FIX: find a more precise way of closing the image upload modal
+
+    //add source code files
+    cy.get('[data-cy="upload-code"]')
+      .first()
+      .selectFile("cypress/fixtures/codebase/testSourceCode.txt", {
+        force: true,
+      });
+    cy.get('[data-cy="upload-docs"]')
+      .first()
+      .selectFile("cypress/fixtures/codebase/testNarrativeDocumentation.txt", {
+        force: true,
+      });
+    cy.get('[data-cy="upload-data"]')
+      .first()
+      .selectFile("cypress/fixtures/codebase/testUploadData.txt", {
+        force: true,
+      });
+    cy.get('[data-cy="upload-results"]')
+      .first()
+      .selectFile("cypress/fixtures/codebase/testSimulationOutput.txt", {
+        force: true,
+      });
+    cy.get('[data-cy="add metadata"]').click();
+    cy.get('[data-cy="release-notes"] textarea').type("Release notes");
+    cy.get('[data-cy="embargo-end-date"]').click();
+    cy.get('[data-cy="embargo-end-date"]').contains("29").click();
+    cy.get('[data-cy="operating-system"] select').select(
+      "Operating System Independent"
+    );
+    cy.get('[data-cy="software-frameworks"]').type("NetLogo {enter} ");
+    cy.get("body").click(0, 0);
+    cy.get('[data-cy="programming-languages"').type("Net Logo {enter}");
+    cy.get("body").click(0, 0);
+    cy.get('[data-cy="license"] .multiselect__select').click();
+    cy.get('[data-cy="license"] .multiselect__element')
+      .contains("GPL-2.0")
+      .click();
+    cy.get('[data-cy="save-and-continue"]').click();
+    cy.get('button.btn.btn-danger[rel="nofollow"]').click();
+    cy.get('button[type="submit"].btn.btn-danger[form="publish-form"]').click();
+  });
+
+  it("should verify that the codebase was uploaded correctly", () => {
+    cy.visit("/codebases");
+    cy.get(".search-result").first().find("a").first().click();
+    assert(cy.get("h1").contains("Codebase Title"));
```

Comment:
These kinds of assertions are highly dependent on whatever stylistic or textual changes we want to make in the frontend. It might be better to have a list or other data structure of expected text and programmatically iterate through it with assertions and comment the data structure as a place of possible future change.

Reply:
Good idea, perhaps a global object that describes the initial database state which can be modified at runtime? for example:

```js
{
  ...
  codebases: [
    { title: "Codebase Title", description: "something" },
  ],
  users: [
    { username: "test_user", password: "123456", ... }
  ],
  ...
}
``` 

and then, as Allen said, use this to assert that certain things are found on a page, rather than at some specific location in the DOM tree